File: 剑指offer/二维数组中的查找/src/cpp/二维数组中的查找.cpp
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        bool find=false;
        for(int row=0,col=array[row].size()-1;row < array.size()&&col>=0;){
                if (target==array[row][col])
                    return find=true;
                else if(array[row][col]>target)
                    col--;
                else
                    row++;
        }
        return find;
    }
};

File: 剑指offer/二维数组中的查找/src/python/二维数组中的查找.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        isFind='false';
        row=0
        col=len(array[0])-1
        while (row<len(array))and(col>=0):
            if target==array[row][col]:
                isfind='true'
                return isfind
            elif array[row][col]>target:
                col=col-1
            else :
                row=row+1
        return isFind

while True:
    try:
        number=input()
        a=number[0]
        b=number[1]
        f=Solution()
        print f.Find(a,b)
    except:
        break